packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -187,14 +191,40 @@ const App = (props) => {
                 body: {}
             })
         }
+    }, [baskets])
+
+    useEffect(() => {
         // update the basket currency if it doesn't match the current locale currency
-        if (baskets?.baskets?.[0]?.currency && baskets.baskets[0].currency !== currency) {
+        if (basket?.currency && basket?.currency !== currency) {
```

Comment:
I moved the basket currency update away from createBasket and put it together with the update email in a separate useEffect.

This allows us to update a basket even if the list of baskets aren't changed (and because I was getting the too many baskets error when I have createBasket grouped with the email update)

Reply:
Update: currency and email updates have been separated into 2 separate useEffects